main/src/ca/uwaterloo/flix/language/ast/Type.scala

```diff
@@ -397,6 +383,25 @@ object Type {
     }
   }
 
+  /**
+    * Extends the record row `rest` with the given type and label.
+    */
+  def mkExtendedRecordRow(label: String, tpe: Type, rest: Type): Type = {
+    val extendedRow = mkApply(Type.Cst(TypeConstructor.ExtendedRecordRow(label)), List(tpe, rest))
+    Type.Apply(Type.Cst(TypeConstructor.Record), extendedRow)
+  }
+
+  // MATT this and the above method form a sort of wrapper case class thing. Is there a better way to do this?
+  object matchExtendedRecordRow {
```

Comment:
We generally do not use custom pattern matching, since I think it is preferable to see what is going on, even if it is more verbose. In the rare cases where we do have custom pattern matching, we keep it local to that phase or module.